### DISCUSSION TITLE
How to use gradient similarity to guide diffusion models?

### DISCUSSION THREAD

**Discussion by Toki912** [huggingface/diffusers#9456] [Q&A]
Here is the English translation of your summarized problem scenario:

1) An image A is passed through a ResNet network to compute its gradient \( g \);
2) The intermediate image \( x_t \) during the diffusion model's generation process;
3) The gradient \( g_t \) of \( x_t \) is computed using the same ResNet network;
4) The cosine similarity between \( |g_t - g| \) is calculated as guidance information, which is used to adjust the mean and variance of the diffusion model's generation process;
5) When the diffusion model iterates to \( t = 0 \), as \( g_t \) (for \( t == 0 \)) approaches \( g \), the generated image \( x_0 \) by the diffusion model also becomes highly consistent with image A.

**Comment by asomoza** [comment]
this isn't an issue and you already posted the same in the discussion which is the correct place, closing this from now. If you encounter a problem related to the diffusers library please feel free to open an issue but for questions and help please use the discussion. 

Also please avoid opening multiple issues or discussions about the same topic, doing this won't get you better or faster answers.
